### ISSUE TITLE
'cudaStreamSynchronize(d->stream)' failed: an illegal memory access was encountered

### ISSUE THREAD

**Issue by Selur** [AmusementClub/vs-fgrain-cuda#2]
using:
```
# Imports
import vapoursynth as vs
# getting Vapoursynth core
core = vs.core
# Loading Plugins
core.std.LoadPlugin(path="i:/Hybrid/64bit/vsfilters/SourceFilter/DGDecNV/DGDecodeNV.dll")
core.std.LoadPlugin(path="C:/Users/Selur/Desktop/fgrain_cuda.dll")
# source: 'G:\TestClips&Co\files\Drone_rocky.mov'
# current color space: YUV420P8, bit depth: 8, resolution: 3840x2160, fps: 23.976, color matrix: 709, yuv luminance scale: limited, scanorder: progressive
# Loading G:\TestClips&Co\files\Drone_rocky.mov using DGSource
clip = core.dgdecodenv.DGSource("E:/Temp/mov_41e809d72df16311ce1943fa427c83ff_853323747.dgi")
# Setting color matrix to 709.
clip = core.std.SetFrameProps(clip, _Matrix=1)
clip = clip if not core.text.FrameProps(clip,'_Transfer') else core.std.SetFrameProps(clip, _Transfer=1)
clip = clip if not core.text.FrameProps(clip,'_Primaries') else core.std.SetFrameProps(clip, _Primaries=1)
# Setting color range to TV (limited) range.
clip = core.std.SetFrameProp(clip=clip, prop="_ColorRange", intval=1)

clip = core.resize.Bicubic(clip=clip, width=1280, height=720, format=vs.GRAYS)
# core.fgrain_cuda.Add(clip clip[, int num_iterations = 800, float grain_radius_mean = 0.1, float grain_radius_std=0.0, float sigma = 0.8, int seed = 0])
clip = core.fgrain_cuda.Add(clip)

# making sure frame rate is set to 23.976
clip = core.std.AssumeFPS(clip=clip, fpsnum=24000, fpsden=1001)
# Output
clip.set_output()
```
I get:
`'cudaStreamSynchronize(d->stream)' failed: an illegal memory access was encountered`
I first thought this might be a conflict with dgdecodenv, but using LibavSMASHSource the same error occurs.

Works when using: 
- width=720, height=480
- width=960, height= 544
fails when using:
- width=1024, height=576
vram usage is always around 2GB.

Using a Geforce GTX 1070ti with 512.15 driver on Windows 11 in case it matters.

**Comment by Selur** [comment]
New commit seems to fix the issue.  👍 
````
clip = core.resize.Bicubic(clip=clip, width=1920, height=1080)
Y, U, V = core.std.SplitPlanes(clip)
Y = core.resize.Bicubic(clip=Y, format=vs.GRAYS)
# core.fgrain_cuda.Add(clip clip[, int num_iterations = 800, float grain_radius_mean = 0.1, float grain_radius_std=0.0, float sigma = 0.8, int seed = 0])
Y = core.fgrain_cuda.Add(Y)
Y = core.resize.Bicubic(clip=Y, format=vs.GRAY8)
clip = core.std.ShufflePlanes(clips=[Y, U, V], planes=[0, 0, 0], colorfamily=vs.YUV)
````
works fine.

**Comment by AkarinVS** [comment]
Fixed by c16db618f55f9e69ded4125007b89ef84062b385.
